File: testdata/jennies/builders/map_index_assignment/GoBuilder/sandbox/somestruct_builder_gen.go
package sandbox

import (
	cog "github.com/grafana/cog/generated/cog"
)

var _ cog.Builder[SomeStruct] = (*SomeStructBuilder)(nil)

type SomeStructBuilder struct {
    internal *SomeStruct
    errors cog.BuildErrors
}

func NewSomeStructBuilder() *SomeStructBuilder {
	resource := NewSomeStruct()
	builder := &SomeStructBuilder{
		internal: resource,
		errors: make(cog.BuildErrors, 0),
	}

	return builder
}



func (builder *SomeStructBuilder) Build() (SomeStruct, error) {
	if err := builder.internal.Validate(); err != nil {
		return SomeStruct{}, err
	}
	
	if len(builder.errors) > 0 {
	    return SomeStruct{}, cog.MakeBuildErrors("sandbox.someStruct", builder.errors)
	}

	return *builder.internal, nil
}

func (builder *SomeStructBuilder) Annotations(key string,value string) *SomeStructBuilder {
if builder.internal.Annotations == nil {
    builder.internal.Annotations = map[string]string{}
}
    builder.internal.Annotations[key] = value

    return builder
}

File: testdata/jennies/builders/map_index_assignment/TypescriptBuilder/src/sandbox/someStructBuilder.gen.ts
import * as cog from '../cog';
import * as sandbox from '../sandbox';

export class SomeStructBuilder implements cog.Builder<sandbox.SomeStruct> {
    protected readonly internal: sandbox.SomeStruct;

    constructor() {
        this.internal = sandbox.defaultSomeStruct();
    }

    /**
     * Builds the object.
     */
    build(): sandbox.SomeStruct {
        return this.internal;
    }

    annotations(key: string,value: string): this {
        if (!this.internal.annotations) {
            this.internal.annotations = {};
        }
        this.internal.annotations[key] = value;
        return this;
    }
}

